var/spack/repos/builtin/packages/blast-plus/package.py

```diff
@@ -81,6 +81,7 @@ def patch(self):
     depends_on("pcre", when="+pcre")
 
     depends_on("python", when="+python")
+    depends_on("py-setuptools", when="+python", type="build")
```

Comment:
Since this is due to distutils removal, we need a setuptools with distutils, and only when python is new enough:
```suggestion
    depends_on("py-setuptools@48:", when="+python ^python@3.12:", type="build")
```